src/log/Entry.h

```diff
@@ -4,103 +4,112 @@
 #ifndef __CEPH_LOG_ENTRY_H
 #define __CEPH_LOG_ENTRY_H
 
-#include "common/CachedPrebufferedStreambuf.h"
-#include <pthread.h>
-#include <string>
 #include "log/LogClock.h"
 
+#include "common/StackStringStream.h"
 
-namespace ceph {
-namespace logging {
+#include "boost/container/small_vector.hpp"
 
-struct Entry {
-  log_time m_stamp;
-  pthread_t m_thread;
-  short m_prio, m_subsys;
-  Entry *m_next;
+#include <pthread.h>
 
-  size_t m_buf_len;
-  size_t* m_exp_len;
-  char m_static_buf[1];
+#include <string_view>
 
-  prebuffered_data m_data;
-  CachedPrebufferedStreambuf* m_streambuf;
+namespace ceph {
+namespace logging {
 
-  Entry()
-    : Entry(log_time{}, 0, 0, 0, nullptr)
-  {}
-  Entry(log_time s, pthread_t t, short pr, short sub,
-	const char *msg = NULL)
-    : Entry(s, t, pr, sub, m_static_buf, sizeof(m_static_buf), nullptr,
-	    msg)
+class Entry {
+public:
+  using time = log_time;
+
+  Entry() = delete;
+  Entry(short pr, short sub) :
+    m_stamp(clock().now()),
```

Comment:
`now()` is a static function, so `log_clock::now()` would work without a helper function

Reply:
I tried that but it doesn't work:

````
In file included from /home/pdonnell/ceph/src/log/Log.h:19:0,
                 from /home/pdonnell/ceph/src/common/dout.h:26,
                 from /home/pdonnell/ceph/src/common/debug.h:18,
                 from /home/pdonnell/ceph/src/erasure-code/shec/ErasureCodePluginShec.cc:22:
/home/pdonnell/ceph/src/log/Entry.h: In constructor ‘ceph::logging::Entry::Entry(short int, short int)’:
/home/pdonnell/ceph/src/log/Entry.h:26:28: error: cannot call member function ‘ceph::logging::log_clock::time_point ceph::logging::log_clock::now()’ without object
     m_stamp(log_clock::now()),
                            ^
````

The method is a trick to have a `static log_clock` without requiring a definition in a corresponding `Entry.cc` file. Do you think it's not worth doing?